api/host.py

```diff
@@ -147,7 +157,27 @@ def delete_host_list(
     tags=None,
     filter=None,
 ):
-    if not any([display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, tags, filter]):
+    input_args = deepcopy(locals())
+    args_count = _get_input_args_count(input_args)
+
+    if not args_count:
+        logger.error("bulk-delete operation needs at least one input property to filter on.")
+        flask.abort(400, "bulk-delete operation needs at least one input property to filter on.")
+
+    if not any(
+        [
+            display_name,
+            fqdn,
+            hostname_or_id,
+            insights_id,
+            provider_id,
+            provider_type,
+            registered_with,
+            staleness,
+            tags,
+            filter,
+        ]
+    ):
         logger.error("bulk-delete operation needs at least one input property to filter on.")
         flask.abort(400, "bulk-delete operation needs at least one input property to filter on.")
```

Comment:
I'm curious why the addition of args_count is necessary when it seems like it checks the same thing as the `if not any` statement. 

If I'm wrong there and it is necessary to have both though I'd prefer to have them in an or to avoid repeating the error and abort calls. I also slightly prefer using `== 0` to `not`, it just feels a little more descriptive to me, or fitting of the check being done, but I won't block or insist on it.
```suggestion
    if args_count == 0 or if not any(
        [
            display_name,
            fqdn,
            hostname_or_id,
            insights_id,
            provider_id,
            provider_type,
            registered_with,
            staleness,
            tags,
            filter,
        ]
    ):
        logger.error("bulk-delete operation needs at least one input property to filter on.")
        flask.abort(400, "bulk-delete operation needs at least one input property to filter on.")
```

Reply:
I agree. In addition, we've already got all the args in the variable `input_args` at this point; instead of listing them all out again, you can just do `if not any( [*input_args] )`.

Reply:
> I'm curious why the addition of args_count is necessary when it seems like it checks the same thing as the if not any statement.
@ryandillinfelton `args_count()` checks if any parameter with non-None value has been passed in.  `if not any([...]) could not tell me.  i.e. counts the number of arguments passed in.

Reply:
@kruai checking of number arguments passed in is gone because `if not any([...])` does both jobs of checking for `None` values and no args passed.